lib/src/customer_io.dart

```diff
@@ -0,0 +1,347 @@
+      if (propsConfig != null) {
+        _configurations = propsConfig;
+        developer.log(
+            'Customer.io SDK initialized from properties in $pathToSDKCredentials');
+      } else {
+        // initializing with dummy values to avoid unwanted runtime exceptions
+        // on configurations
+        _configurations =
+            CustomerIOConfigurations(siteId: 'siteId', apiKey: 'apiKey');
+        developer.log('Customer.io SDK could not be initialized properly');
+        return Future.error(Exception(
+            'No values found for Customer.io SDK in preferences or $pathToSDKCredentials'));
+      }
+    }
+
+    return CustomerIO.initialize(
+      config: CustomerIOConfig(
+        siteId: _configurations.siteId,
+        apiKey: _configurations.apiKey,
+        organizationId: _configurations.organizationId ?? '',
+        region: _configurations.region ?? Region.us,
+      ),
+    ).then((value) => _platform.invokeMethod('captureLogs'));
+  }
+
+  /// Saves profile identifier locally to identify login state and triggers
+  /// SDK call for profile identification
+  Future<bool> saveProfileIdentifier(String identifier) =>
+      SharedPreferences.getInstance().then((prefs) {
+        return prefs.setString(_profileIdentifier, identifier);
+      });
+
+  Future<String?> fetchProfileIdentifier() =>
+      SharedPreferences.getInstance().then((prefs) {
+        return prefs.getString(_profileIdentifier);
+      });
+
+  Future<bool> clearProfileIdentifier() =>
+      SharedPreferences.getInstance().then((prefs) {
+        return prefs.remove(_profileIdentifier);
+      });
+
+  Future<List<String>?> getLogs() async {
+    try {
+      final List<Object?> result = await _platform.invokeMethod('getLogs');
+      return result
+          .map((log) => log?.toString() ?? 'NULL')
+          .toList(growable: false);
+    } on PlatformException catch (ex) {
+      developer.log("Failed to get logs from SDK: '${ex.message}'", error: ex);
+      return null;
+    }
+  }
+
+  Future<String?> getUserAgent() async {
+    try {
+      final result = await _platform.invokeMethod('getUserAgent');
+      return result?.toString();
+    } on PlatformException catch (ex) {
+      developer.log("Failed to get user agent from SDK: '${ex.message}'",
+          error: ex);
+      return null;
+    }
+  }
+
+  Future<String?> getDeviceToken() async {
+    try {
+      final result = await _platform.invokeMethod('getDeviceToken');
+      return result?.toString();
+    } on PlatformException catch (ex) {
+      developer.log("Failed to get device token from SDK: '${ex.message}'",
+          error: ex);
+      return null;
+    }
+  }
+}
+
+class CustomerIOConfigurations {
+  String siteId;
+  String apiKey;
+  String? organizationId;
+  Region? region;
+  String? trackingUrl;
+  String? gistEnvironment;
+  int? backgroundQueueSecondsDelay;
+  int? backgroundQueueMinNumberOfTasks;
+  bool? featureEnablePush;
```

Comment:
what does this config do?

Reply:
I copied it from designs initially, but I don't think we need this, will remove it in next PR.